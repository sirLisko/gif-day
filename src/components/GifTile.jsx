import React from "react";
import PropTypes from "prop-types";

const GifTile = ({ gif }) => (
  <video key={gif} loop autoPlay muted playsInline>
    <source src={gif} type="video/mp4" />
  </video>
);

GifTile.propTypes = {
  gif: PropTypes.string.isRequired,
};

export default GifTile;
