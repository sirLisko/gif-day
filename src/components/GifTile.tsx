import React from "react";
import { Image } from "types";

interface Props {
  gifObj: Image;
  dynamic?: boolean;
}

const GifTile = ({ gifObj, dynamic }: Props) =>
  gifObj ? (
    <div>
      {dynamic && <img src={gifObj.gif.still} alt={gifObj.text} />}
      <video
        key={gifObj.gif.gif}
        title={gifObj.text}
        loop
        autoPlay
        muted
        playsInline
      >
        <source src={gifObj.gif.gif} type="video/mp4" />
      </video>
    </div>
  ) : (
    <></>
  );

export default GifTile;
