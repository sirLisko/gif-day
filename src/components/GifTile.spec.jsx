import React from 'react';
import { shallow } from 'enzyme';

import GifTile from './GifTile';

describe('GifTile Component', () => {
  let wrapper;
  const props = {
    gifObj: {
      gif: 'foo',
      still: 'bar',
    },
  };

  beforeEach(() => {
    wrapper = shallow(<GifTile {...props} />);
  });

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly - dynamic', () => {
    wrapper.setProps({ dynamic: true });
    expect(wrapper).toMatchSnapshot();
  });
});
