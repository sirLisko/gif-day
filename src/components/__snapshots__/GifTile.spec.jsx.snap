// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifTile Component should render properly 1`] = `
<video
  autoPlay={true}
  key="foo"
  loop={true}
  muted={true}
  playsInline={true}
>
  <source
    src="foo"
    type="video/mp4"
  />
</video>
`;
